repo-agent/repowatch/controllers/repowatch_controller.go

```diff
@@ -117,26 +117,34 @@ func (r *RepoWatchReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 func (r *RepoWatchReconciler) reconcileReviews(ctx context.Context, repoWatch *reviewv1alpha1.RepoWatch, client *github.Client, owner string, repo string) error {
 	log := log.FromContext(ctx)
 
-	// Get open PRs
-	prs, _, err := client.PullRequests.List(ctx, owner, repo, &github.PullRequestListOptions{State: "open"})
-	if err != nil {
-		log.Error(err, "unable to list pull requests")
-		return err
-	}
-
-	// If the review spec has a list of PRs, filter the PRs
+	var prs []*github.PullRequest
 	if len(repoWatch.Spec.Review.PullRequests) > 0 {
-		var filteredPRs []*github.PullRequest
-		for _, pr := range prs {
-			for _, prNumber := range repoWatch.Spec.Review.PullRequests {
-				if *pr.Number == prNumber {
-					filteredPRs = append(filteredPRs, pr)
-					break
-				}
+		// If specific PRs are requested, fetch them directly
+		for _, prNumber := range repoWatch.Spec.Review.PullRequests {
+			pr, _, err := client.PullRequests.Get(ctx, owner, repo, prNumber)
+			if err != nil {
+				log.Error(err, "unable to get pull request", "prNumber", prNumber)
```

Comment:
I'd like to consider how we better surface this. Logging is a start. However I feel like we either need to be able to push some sort of message to let us know or possibly have a page which lets the user know they have PRs we were unable to process.